crates/goose-cli/src/commands/session.rs

```diff
@@ -17,11 +17,35 @@ use crate::prompt::thinking::get_random_goose_action;
 use crate::session::session::Session;
 use crate::session::session_file::ensure_session_dir;
 
-pub fn build_session<'a>(session: Option<String>, profile: Option<String>) -> Box<Session<'a>> {
-    let session_name = session_name(session);
-    let session_file: PathBuf = ensure_session_dir()
-        .expect("Failed to create session directory")
-        .join(format!("{}.jsonl", session_name));
+pub fn build_session<'a>(
+    session: Option<String>,
+    profile: Option<String>,
+    resume: bool,
+) -> Box<Session<'a>> {
+    let session_dir = ensure_session_dir().expect("Failed to create session directory");
+
+    let session_name = if session.is_none() && !resume {
+        generate_new_session_name(&session_dir)
+    } else {
+        session_name_or_rand(session.clone())
```

Comment:
nit: can be a later PR but if resume is set and no name is specified i think we would lookup the most recent session here?